src/overlay/Peer.h

```diff
@@ -337,6 +336,15 @@ class Peer : public std::enable_shared_from_this<Peer>,
 
     bool mShuttingDown{false};
 
+#ifdef BUILD_TESTS
+    // For testing purposes, sometimes we want to force
+    // to set or not set the pull mode flag in the auth
+    // message. Set this optional bool value to force it.
+    // This will be obsolete once the min overlay version
+    // becomes >= FIRST_VERSION_REQUIRING_PULL_MODE.
+    std::optional<bool> mMaybeOverwritePullModeFlagForTesting;
```

Comment:
Setting this to true has no effect since pull mode is already required by default, so this should just be `bool mOverrideDisablePullModeForTesting` which defaults to false, and can be set to true.

Reply:
If it's a bool, i don't think we can test the case when the peer is using an old overlay version and wants to turn on pull mode. 

- A: Old overlay version that wants to turn on pull mode
- B: The latest overlay version that always turns on pull mode

Then, in this test, A will see that B uses the latest overlay version that turns on pull mode, and will not turn on the flag. Therefore, B will receive an `AUTH` message without the pull mode flag from an old overlay version, so B will refuse to connect.

Reply:
that's because the "old version" really is running the new code from this PR, so it creates a discrepancy. In reality, old version will set the flag either way. This can be addressed by changing the condition in sendAuth to `if (mRemoteOverlayVersion < Peer::FIRST_VERSION_REQUIRING_PULL_MODE) || mOverlayVersion < Peer::FIRST_VERSION_REQUIRING_PULL_MODE`

Reply:
I don't think it's reproducing old version behavior, as this condition is basically a no-op during normal operation. It makes things a bit simpler since we'll just request pull mode always by default, then in the macro you can disable it if needed.

Reply:
Done

Reply:
Thanks!